addons/account/models/account_payment.py

```diff
@@ -640,11 +682,11 @@ def _synchronize_from_moves(self, changed_fields):
 
                 if len(liquidity_lines) != 1 or len(counterpart_lines) != 1:
                     raise UserError(_(
-                        "The journal entry %s reached an invalid state relative to its payment.\n"
-                        "To be consistent, the journal entry must always contains:\n"
-                        "- one journal item involving the outstanding payment/receipts account.\n"
-                        "- one journal item involving a receivable/payable account.\n"
-                        "- optional journal items, all sharing the same account.\n\n"
+                        "Journal Entry %s is not valid. In order to proceed the journal items must include the following:\n"
+                        "- An outstanding payments/receipts account\n"
+                        "- A receivable/payable account (with an exception of internal transfers)\n"
+                        "- Share the same Partner\n"
+                        "- Share the same account if the journal items are optional\n\n"
```

Comment:
That last line doesn't make sense to me

Reply:
```Journal Entry %s is not valid. In order to proceed the journal items must:
- Include one outstanding payments/receipts account
- Include one receivable/payable account (with an exception of internal transfers)
- Share the same partner
- Share the same account if they are write-offs```
Better ? @tsb-odoo ?

Reply:
@william-andre @jbw-odoo The initial proposition has been discussed with RSH. I mean, the wording is "reviewed" by a (nice) native-speaker. About the deep meaning though. It also didn't appear clear to me and that is why I wanted the error message to be improved. 

Functionally speaking, a payment can have optional items (typically for write-offs) but they must share the same account. You might ask yourself why the user would write multiple journal items on the same account and not group them in one. I guess it could happen to have distinct labels.

Reply:
Only the write-offs should share the same account, otherwise the combination with the first two lines is impossible. The wording (or at least the meaning) before this PR was correct

Reply:
Agreed with TSB
> Journal Entry %s is not valid. In order to proceed the journal items must include the following:
> - An outstanding payments/receipts account
> - A receivable/payable account (with an exception of internal transfers)
> - Share the same Partner
> They can also include optional journal items, all sharing the same account.

But...
> je trouve que c'est toujours super bavard comme message d'erreur et j'aurais préféré qu'on ne trigger que ce qui déconne dans le payment au moment de le poster
donc j'aurais idéalement découpé le message pour qu'il soit déclenché à bon escient